### PR TITLE
Add documentation around our use of pg-bouncer [ci-skip]

### PR DESCRIPTION
**Pull request by vaidehijoshi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

I learned about `pgbouncer` (apparently that's a thing!) in my technical walk-through with @benhalpern, and thought it would be nice to add it to our docs. Not entirely sure that _this_ is the right place for it, so if you disagree, just let me know :)

While learning about PgBouncer, I read that PostgreSQL's default connection limit is set to 100 concurrent connections; does anyone know if we use the default connection limit or if we use PgBouncer to change that? If it's relevant, I can add it to the docs here.

Also cleaned up some formatting in the docs for the page that I modified (well, VSCode cleaned it up if we wanna be specific about it 👀)

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

### CODE REVIEW

File: docs/installation/postgresql.md [hunk 1]
```diff
@@ -46,18 +45,32 @@ DATABASE_URL: postgresql://USERNAME:PASSWORD@localhost
...
+We use [PgBouncer](http://www.pgbouncer.org) to manage connection pooling.
+Database pooling creates a shared pool of connections to our database, rather
+than creating new connection each time. PgBouncer is a wrapper around our
+database connection and ensures that we only have a finite set of "cached"
+connections to the database. This means that our app doesn't need to actually
+connect to the database, it only needs to connect to the pool of connections.
+
+Check out the
+[Procfile](https://github.com/thepracticaldev/dev.to/blob/master/Procfile) to
+see when `pg-bouncer` is started.
```
[the remark this answers is not in the document]
Comment (vaidehijoshi) [reply docs/installation/postgresql.md]: woops! good catch :)

**Comment by vaidehijoshi** [comment]
@rhymes I agree with both your suggestions (thank you)! Since it's likely we could change plans, I'll just link to the Heroku Postgres Plans page and point out that the number of connections is dependent on that.

**Comment by joshpuetz** [comment]
Perhaps operational stuff like this could go in a 'running your own instance' section of the docs (if we want to share), or in Notion if it's something we want to keep internal.

As more people fork and spin up their own instances of the DEV platform, I can see sharing the DevOps lessons we learned being valuable!

**Comment by jacobherrington** [comment]
@joshpuetz I agree that there might another home for this, but for now I think just writing it down is a win. :smile: How does that sound to you?

**Pull request merged**